### PR TITLE
Fix default merging of resource attributes from environment variable

### LINKED ISSUE #1689: The SDK MUST extract information from the OTEL_RESOURCE_ATTRIBUTES environment variable
Currently we do not default to including attributes specified with the `OTEL_RESOURCE_ATTRIBUTES` environment variable. This is required by [the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.0.1/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable):

> The SDK MUST extract information from the `OTEL_RESOURCE_ATTRIBUTES` environment variable and [merge](#merge) this, as the secondary resource, with any resource information provided by the user, i.e. the user provided resource information has higher priority.
> 
> The `OTEL_RESOURCE_ATTRIBUTES` environment variable will contain of a list of key value pairs, and these are expected to be represented in a format matching to the [W3C Baggage](https://github.com/w3c/baggage/blob/fdc7a5c4f4a31ba2a36717541055e551c2b032e4/baggage/HTTP_HEADER_FORMAT.md#header-content), except that additional semi-colon delimited metadata is not supported, i.e.: `key1=value1,key2=value2`. All attribute values MUST be considered strings.

These attributes will be used to create a Resource if [`resource.New`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#New) is used to create a resource passed to an SDK, but it must be done by the user.

The `TracerProvider` and metric `Controller` need to be updated to default extracting these attributes into a resource that is merged with the `resource.Default` resource they already use.

Having these attributes automatically merged into the used resource makes the [`FromEnv` detector](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#FromEnv) of limited value. It would only include attributes that would latter be merged in. It seems like that detector should be removed, or possibly changed to accept a list of environment variable names that it would then pull in as attributes of a Resource. Meaning if you wanted to specify a Resource in a way other than defined for the `OTEL_RESOURCE_ATTRIBUTES` environment variable you could accomplish it that way.

### PR DESCRIPTION
**Pull request by dhruv-vora**
Extracted the resource attributes from the `OTEL_RESOURCE_ATTRIBUTES` environment variable and merged them with the `resource.Default` resource as well as resources provided to the `TracerProvider` and metric `Controller`.

Fixes #1689

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: sdk/resource/resource.go [hunk 1]
```diff
@@ -144,6 +144,17 @@ func Default() *Resource {
 	return defaultResource
 }
 
+// Environment returns an instance of Resource with attributes
+// extracted from OTEL_RESOURCE_ATTRIBUTES environment variable
```
Comment (MrAlias) [inline sdk/resource/resource.go:148]:
```suggestion
// extracted from the OTEL_RESOURCE_ATTRIBUTES environment variable.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ## [Unreleased]
 
 ### Added
-
+- Extract resource attributes from the `OTEL_RESOURCE_ATTRIBUTES` environment variable and merge them with the `resource.Default` resource as well as resources provided to the `TracerProvider` and metric `Controller`. (#1689)
```
Comment (MrAlias) [inline CHANGELOG.md:12]:
```suggestion
- Extract resource attributes from the `OTEL_RESOURCE_ATTRIBUTES` environment variable and merge them with the `resource.Default` resource as well as resources provided to the `TracerProvider` and metric `Controller`. (#1785)
```

File: sdk/trace/provider.go [hunk 3]
```diff
@@ -263,7 +263,9 @@ func WithSpanProcessor(sp SpanProcessor) TracerProviderOption {
 func WithResource(r *resource.Resource) TracerProviderOption {
 	return func(opts *TracerProviderConfig) {
 		if r != nil {
```
Comment (MrAlias) [inline sdk/trace/provider.go:265]: No need for the conditional here, `resource.Merge(resource.Environment(), r)` will handle `r` being `nil`, similar to what was done in the metrics `Controller`.

**Review by MrAlias — APPROVED** [review]

File: sdk/trace/provider.go [hunk 4]
```diff
@@ -262,9 +262,7 @@ func WithSpanProcessor(sp SpanProcessor) TracerProviderOption {
 // resource.Default() Resource by default.
 func WithResource(r *resource.Resource) TracerProviderOption {
 	return func(opts *TracerProviderConfig) {
-		if r != nil {
-			opts.resource = r
-		}
+		opts.resource = resource.Environment()
```
Comment (MrAlias) [inline sdk/trace/provider.go:265]:
```suggestion
		opts.resource = resource.Merge(resource.Environment(), r)
```

**Comment by MrAlias** [comment]
This looks ready to merge. @dhruv-vora can you sync with `main` or allow maintainers to modify the PR?

**Pull request merged**